### ISSUE TITLE
[iOS] NavigationPage.Popped called twice when Navigation.PopAsync is called

### ISSUE THREAD

**Issue by jenscski** [xamarin/Xamarin.Forms#3509]
### Steps to Reproduce

1. Create new app with the following code as the app

```
using System;
using Xamarin.Forms;

namespace PagesGallery
{
	public class App : Xamarin.Forms.Application
	{
		public App()
		{
			var page = new NavigationPage(new TestPage(false));
			page.Popped += (s, e) =>
			{
				(e.Page as TestPage).Popped();
			};

			MainPage = page;
		}
	}

	public class TestPage : ContentPage
	{
		bool _popped = false;

		public TestPage(bool showPop)
		{
			Title = "Test page";

			var content = new StackLayout();

			content.Children.Add(new Button
			{
				Text = "Push page",
				Command = new Command(async () => await Navigation.PushAsync(new TestPage(true), false)),
			});

			if (showPop)
			{
				content.Children.Add(new Button
				{
					Text = "Pop page",
					Command = new Command(async () => await Navigation.PopAsync(false)),
				});
			}

			Content = content;
		}

		internal void Popped()
		{
			if (_popped) throw new Exception("Already popped");

			_popped = true;
		}
	}
}

```
2. On first page press Push page, on next page press Pop Page
3. It should crash because of this ```throw new Exception("Already popped");```

### Expected Behavior
NavigationPage.Popped only called once on each page removed from navigation stack
### Actual Behavior
NavigationPage.Popped called twice when Navigation.PopAsync is used to remove page from navigation stack
### Basic Information

- Version with issue: 3.1.0.637273
- Last known good version: 3.0 and 3.2.0.637442-pre1
- IDE: VS 15.7.6
- Platform Target Frameworks:
  - iOS:  11.12.0.4
- Nuget Packages:
- Affected Devices:

**Comment by TimLariviere** [comment]
Encountered the same bug.
But I ran the repro with XF 3.0.0.561731 and XF 3.2.0.637442-pre1, this is not specific to XF 3.1.
The problem was already here and is still here.

This bug is especially bad when using [Elmish.XamarinForms](https://github.com/fsprojects/Elmish.XamarinForms) as we need to manage our own page stack but can't keep in sync with NavigationPage on iOS because of this.

I investigated a little and found that when calling PopAsync, 2 things happened.
1. PopAsync raises the Popped event (this is good)
2. The `ParentingViewController` of the already removed page is called by `DidMoveToParentViewController` which in turn will call `UpdateFormsInnerNavigation` which will notify the NavigationPage to raise a second Popped event. (this is bad)

`NavigationRenderer` protects itself from this double event but setting a `_ignorePopCall` flag.
https://github.com/xamarin/Xamarin.Forms/blob/e3b6e4755eda0e7611694929d6b0529ee2c40d4b/Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs#L842-L851
